src/Config.php

```diff
@@ -417,38 +423,111 @@ public function checkIgnored($payload, $accessToken, $toLog, $isUncaught)
+            return $this->shouldIgnoreError($toLog);
+        }
+        
+        if ($toLog instanceof \Exception) {
+            return $this->shouldIgnoreException($toLog);
+        }
+        
+        return false;
+    }
+    
+    /**
+     * Check if the error should be ignored due to `included_errno` config,
+     * `use_error_reporting` config or `error_sample_rates` config.
+     *
+     * @param \Rollbar\ErrorWrapper $toLog
+     *
+     * @return bool
+     */
+    protected function shouldIgnoreError(ErrorWrapper $toLog)
+    {
+        $errno = $toLog->errorLevel;
 
-            if ($this->included_errno != -1 && ($errno & $this->included_errno) != $errno) {
-                // ignore
-                return true;
-            }
+        if ($this->included_errno != -1 && ($errno & $this->included_errno) != $errno) {
+            // ignore
+            return true;
+        }
 
-            if ($this->use_error_reporting && ($errno & error_reporting()) != $errno) {
-                // ignore due to error_reporting level
-                return true;
-            }
+        if ($this->use_error_reporting && ($errno & error_reporting()) != $errno) {
+            // ignore due to error_reporting level
+            return true;
+        }
 
-            if (isset($this->error_sample_rates[$errno])) {
-                // get a float in the range [0, 1)
-                // mt_rand() is inclusive, so add 1 to mt_randmax
-                $float_rand = mt_rand() / ($this->mt_randmax + 1);
-                if ($float_rand > $this->error_sample_rates[$errno]) {
-                    // skip
-                    return true;
-                }
+        if (isset($this->error_sample_rates[$errno])) {
+            // get a float in the range [0, 1)
+            // mt_rand() is inclusive, so add 1 to mt_randmax
+            $float_rand = mt_rand() / ($this->mt_randmax + 1);
+            if ($float_rand > $this->error_sample_rates[$errno]) {
+                // skip
+                return true;
             }
         }
+        
+        return false;
+    }
+    
+    /**
+     * Check if the exception should be ignored due to configured exception
+     * sample rates.
+     *
+     * @param \Exception $toLog
+     *
+     * @return bool
+     */
+    protected function shouldIgnoreException(\Exception $toLog)
+    {
+        // get a float in the range [0, 1)
+        // mt_rand() is inclusive, so add 1 to mt_randmax
+        $floatRand = mt_rand() / ($this->mt_randmax + 1);
+        if ($floatRand > $this->exceptionSampleRate($toLog)) {
+            // skip
+            return true;
+        }
+        
         return false;
     }
+    
+    /**
+     * Calculate what's the chance of logging this exception according to
+     * exception sampling.
+     *
+     * @param \Exception $toLog
+     *
+     * @return float
+     */
+    public function exceptionSampleRate(\Exception $toLog)
+    {
+        $sampleRate = 1.0;
+        
```

Comment:
if `$this->exception_sample_rates` is empty which will be often because that is the default, we should not go through the work of gathering the class hierarchy for this exception, we should just return 1.0 early.